### PR TITLE
<DropdownBase/> - Add isOpen to render function

### PR DESCRIPTION
**Pull request by splintor**
### 🔦 Summary
Add an `isOpen` field to the rendering function of the `<DropdownBase>` component. This can be used to change the rendered content based on whether the items list is currently opened (see [sample use case](https://wix.slack.com/archives/C39FTGUBZ/p1594101111156000)).

### ✅ Checklist

- [X] 👨‍💻 API change is approved by the UI Infra Developers 
- 📚 Change is documented
  - [X] API description
- 🔬 Change is tested
  - [X] Component

### CODE REVIEW

File: src/DropdownBase/test/DropdownBase.spec.js [hunk 1]
```diff
@@ -199,6 +199,28 @@ describe('DropdownBase', () => {
...
+            <IconButton dataHook={targetDataHook} onClick={open}>
+              <ChevronDown />
+            </IconButton>
+          );
+        }}
+      </DropdownBase>,
+    );
+
+    await driver.clickTargetElement(targetDataHook);
+    await driver.isDropdownShown();
```
Comment (EgoziE) [inline src/DropdownBase/test/DropdownBase.spec.js:220]: This function returns `true` if the dropdown is being shown, did you mean to use `expect` here?

File: src/DropdownBase/README.API.md [hunk 2]
```diff
@@ -15,6 +15,7 @@ The `<DropdownBase/>` can accept two types of children:
   | `open(e)` | `Function` | A function that will open the items list. Can be called only when in "Uncontrolled" mode. |
   | `close(e)` | `Function` | A function that will close the items list. Can be called only when in "Uncontrolled" mode. |
   | `toggle(e)` | `Function` | A function that will toggle the items list. Can be called only when in "Uncontrolled" mode. |
+  | `isOpen` | `Boolean` | Indicates whether the items list is currently open.  |
```
Comment (EgoziE) [inline src/DropdownBase/README.API.md:18]: This readme is deprecated and we don't show it anymore
Please add a comment in proptypes section instead

File: src/DropdownBase/test/DropdownBase.spec.js (hunk 1 above)
Comment (splintor) [reply src/DropdownBase/test/DropdownBase.spec.js:220]: No, it is expected in the previous test. I just added it to make sure the component is rendered after the click before the next line. But it is probably not needed - I'll remove it.

File: src/DropdownBase/README.API.md (hunk 2 above)
Comment (splintor) [reply src/DropdownBase/README.API.md:18]: 👍

File: src/DropdownBase/test/DropdownBase.spec.js [hunk 3]
```diff
@@ -199,6 +199,27 @@ describe('DropdownBase', () => {
...
+          return (
+            <IconButton dataHook={targetDataHook} onClick={open}>
+              <ChevronDown />
+            </IconButton>
+          );
+        }}
+      </DropdownBase>,
+    );
+
+    await driver.clickTargetElement(targetDataHook);
```
Comment (EgoziE) [inline src/DropdownBase/test/DropdownBase.spec.js:219]: Check both cases, what do you say?
```suggestion
    expect(isListOpen).toEqual(false);
    await driver.clickTargetElement(targetDataHook);
```

Comment (splintor) [reply src/DropdownBase/test/DropdownBase.spec.js:219]: But initially, it is `undefined`. Do you want to test that it is set back to `false` on a second click?

File: src/DropdownBase/DropdownBase.js [hunk 4]
```diff
@@ -295,7 +296,7 @@ class DropdownBase extends React.PureComponent {
 
   _renderChildren() {
     const { children } = this.props;
-    const { selectedId } = this.state;
+    const { selectedId, open: isOpen } = this.state;
```
Comment (EgoziE) [inline src/DropdownBase/DropdownBase.js:299]: If open can be undefined:
```suggestion
    const { selectedId, open } = this.state;
```

File: src/DropdownBase/DropdownBase.js [hunk 5]
```diff
@@ -307,6 +308,7 @@ class DropdownBase extends React.PureComponent {
           open: this._open,
           close: this._close,
           toggle: this._toggle,
+          isOpen,
```
Comment (EgoziE) [inline src/DropdownBase/DropdownBase.js:311]: If open can be undefined:
```suggestion
          isOpen: Boolean(open),
```

File: src/DropdownBase/test/DropdownBase.spec.js (hunk 3 above)
Comment (EgoziE) [reply src/DropdownBase/test/DropdownBase.spec.js:221]: That's not good, I added you 2 inline suggestions for this case.

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://www.wix.com/pages/wix-style-react.pr_5651/

**Pull request merged**